### PR TITLE
Replace deprecated usage of Java.java_home_cmd

### PR DESCRIPTION
**Pull request by bayandin**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/scm-manager.rb [hunk 1]
```diff
@@ -25,25 +25,13 @@ def install
...
-      BASEDIR="#{libexec}"
-      REPO="#{libexec}/lib"
-      export JAVA_HOME=$(#{Language::Java.java_home_cmd("1.8")})
-      "#{libexec}/bin/scm-server" "$@"
-    EOS
-    chmod 0755, bin/"scm-server"
-
-    tools = libexec/"tools"
-    tools.install resource("client")
+    env = Language::Java.overridable_java_home_env("1.8")
```
Comment (SMillerDev) [inline Formula/scm-manager.rb:28]: Recomends java 11 and is currently at 2.5.0 upstream

File: Formula/gosu.rb [hunk 2]
```diff
@@ -19,8 +19,7 @@ class Gosu < Formula
   skip_clean "libexec/ext"
 
   def install
-    cmd = Language::Java.java_home_cmd("1.8")
-    ENV["JAVA_HOME"] = Utils.safe_popen_read(cmd).chomp
+    ENV["JAVA_HOME"] = Language::Java.java_home("1.8")
```
Comment (SMillerDev) [inline Formula/gosu.rb:22]: This also works with java 11

File: Formula/swagger-codegen@2.rb [hunk 3]
```diff
@@ -23,13 +23,11 @@ class SwaggerCodegenAT2 < Formula
   depends_on java: "1.8"
 
   def install
-    # Need to set JAVA_HOME manually since maven overrides 1.8 with 1.7+
-    cmd = Language::Java.java_home_cmd("1.8")
-    ENV["JAVA_HOME"] = Utils.safe_popen_read(cmd).chomp
+    ENV["JAVA_HOME"] = Language::Java.java_home("1.8")
```
Comment (SMillerDev) [inline Formula/swagger-codegen@2.rb:26]: According to the website openjdk@11 could be supported. They're not very clear about it.

**Comment by SMillerDev** [comment]
https://github.com/Homebrew/homebrew-core/pull/61253 should make the swagger-codegen fix redundant

File: Formula/scm-manager.rb (hunk 1 above)
Comment (bayandin) [reply Formula/scm-manager.rb:28]: It seems they have changed it quite a lot. I would update it in a separate PR

And current version doesn't work with java 11:
```
$ scm-cli-client version
Exception in thread "main" java.lang.NoClassDefFoundError: javax/xml/bind/JAXBException
	at sonia.scm.cli.config.ScmClientConfig.load(ScmClientConfig.java:108)
	at sonia.scm.cli.config.ScmClientConfig.getInstance(ScmClientConfig.java:90)
	at sonia.scm.cli.App.loadConfig(App.java:190)
	at sonia.scm.cli.App.run(App.java:153)
	at sonia.scm.cli.App.main(App.java:127)
Caused by: java.lang.ClassNotFoundException: javax.xml.bind.JAXBException
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:581)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)
	... 5 more
```

**Pull request closed without merging**